File: apps/web/src/auth/authorization.ts
import type { AuthUser } from '../types/auth'

export function hasPermission(usuario: AuthUser | null | undefined, permission: string) {
  return usuario?.permisos?.includes(permission) ?? false
}

export function hasAnyPermission(
  usuario: AuthUser | null | undefined,
  permissions: readonly string[],
) {
  return permissions.some((permission) => hasPermission(usuario, permission))
}

export function hasRole(usuario: AuthUser | null | undefined, role: string) {
  const expected = role.replace(/^ROLE_/, '')
  return usuario?.roles?.some((value) => value.replace(/^ROLE_/, '') === expected) ?? false
}

File: apps/mobile/app/src/main/java/ec/edu/uteq/scli/mobile/features/auth/data/Authorization.kt
package ec.edu.uteq.scli.mobile.features.auth.data

fun AuthUserResponse.hasPermission(permission: String): Boolean = permission in permisos
fun AuthUserResponse.hasAnyPermission(vararg permissions: String): Boolean =
    permissions.any(permisos::contains)
fun AuthUserResponse.hasRole(role: String): Boolean = role in roles
